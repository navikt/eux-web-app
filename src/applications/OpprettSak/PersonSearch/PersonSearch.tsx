import { Alert, Loader, Search } from '@navikt/ds-react'
import { Person } from 'declarations/types'
import _ from 'lodash'
import { PileDiv, VerticalSeparatorDiv } from '@navikt/hoykontrast'
import PT from 'prop-types'
import React, { useCallback, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'

export interface PersonSearchProps {
  alertMessage: JSX.Element | string | undefined
  alertType: string | undefined
  alertTypesWatched: Array<string> | undefined
  className?: string
  error: string | undefined
  id ?: string
  initialFnr: any
  parentNamespace: string
  searchingPerson: boolean
  onFnrChange?: (newFnr: string) => void
  onPersonFound?: (person: Person) => void
  onSearchPerformed: (fnr: any) => void
  person?: Person | null | undefined
  value: string | undefined
}

const PersonSearch: React.FC<PersonSearchProps> = ({
  alertMessage,
  alertType,
  alertTypesWatched = [],
  error,
  id,
  initialFnr,
  parentNamespace,
  searchingPerson,
  onFnrChange,
  onPersonFound,
  onSearchPerformed,
  person,
  value
}: PersonSearchProps): JSX.Element => {
  const { t } = useTranslation()
  const [fnr, setFnr] = useState<string>(initialFnr ?? value)
  const [_person, setPerson] = useState<Person | null | undefined>(undefined)
  const [localValidation, setLocalValidation] = useState<string | undefined>(undefined)
  const namespace = parentNamespace + '-personSearch'

  useEffect(() => {
    setFnr(value ?? initialFnr)
  }, [value])

  const isPersonValid = useCallback((person: Person) =>
    person?.fornavn?.length !== undefined && person?.fnr !== undefined,
  []
  )

  useEffect(() => {
    if (person && !_person && isPersonValid(person)) {
      setPerson(person)
      if (_.isFunction(onPersonFound)) {
        onPersonFound(person)
      }
    }
  }, [person])

  const onSearch = (fnr: string | number | readonly string[]) => {
    if (!fnr) {
      setLocalValidation(t('validation:noFnr'))
      return
    }
    const fnrPattern = /^[0-9]{11}$/
    if (fnr && !fnrPattern.test('' + fnr)) {
      setLocalValidation(t('validation:invalidFnr'))
      return
    }
    setLocalValidation(undefined)
    setPerson(undefined)
    if (_.isFunction(onSearchPerformed)) {
      onSearchPerformed(fnr)
    }
  }

  const onChange = (fnr: string): void => {
    setLocalValidation(undefined)
    const newFnr = fnr.trim()
    setFnr(newFnr)
    if (_.isFunction(onFnrChange)) {
      onFnrChange(newFnr)
    }
  }

  return (
    <PileDiv style={{ alignItems: 'flex-start' }}>
      <Search
        label={t('label:sÃ¸ker')}
        /* error={error ?? localValidation} */
        data-testid={id ?? namespace + '-saksnummerOrFnr'}
        id={id ?? namespace + '-saksnummerOrFnr'}
        onChange={onChange}
        required
        hideLabel={false}
        value={fnr || ''}
        disabled={searchingPerson}
        onSearch={onSearch}
      >
        <Search.Button>
          {searchingPerson ? t('message:loading-searching') : t('el:button-search')}
          {searchingPerson && <Loader />}
        </Search.Button>
      </Search>
      {(error ?? localValidation) && (
        <>
          <VerticalSeparatorDiv size='0.5' />
          <span className='navds-error-message navds-error-message--medium'>
            {error ?? localValidation}
          </span>
        </>
      )}
      {alertMessage && alertType && alertTypesWatched.indexOf(alertType) >= 0 && (
        <Alert variant='warning'>
          {alertMessage}
        </Alert>
      )}
    </PileDiv>
  )
}

PersonSearch.propTypes = {
  className: PT.string,
  onFnrChange: PT.func,
  onPersonFound: PT.func
  // validation: ErrorElementPropType
}

export default PersonSearch
