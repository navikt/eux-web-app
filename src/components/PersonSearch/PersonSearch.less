@import 'src/constants';

.personsok {
  margin-bottom: 2em;

  .personsok__skjema {
    display: flex;
    flex-direction: row;
  }
  .personsok__input {
    min-width: 24.5em;
  }
  .personsok__spinnerwrapper {
    margin: 2.3em 1em 0 -2.5em;
  }

  .personsok__knapp {
    display: flex;
    flex: 0;
    height: 2.4em;
    align-self: flex-start;
    margin: 1.9em 0 0 1em;
  }

  .personsok__kort {
    margin: 1em 0;
  }

  .personsok__advarsel {
    margin-top: 1em;
  }
  .familierelasjoner__knapp {
    display: flex;
    align-self: center;
    justify-self: flex-end;

    .familierelasjoner__knapp__label {
      display: flex;
      align-self: center;
    }
  }
  .familierelasjoner__knapp__ikon {
    margin: -0.0em 0.5em 0 0;

    path {
      stroke: @navBla;
    }
  }
  .knapp--disabled {
    path {
      stroke: @white !important;
    }
  }

  .familierelasjoner__knapp {

    &:hover:not(.knapp--disabled) {
      .familierelasjoner__knapp__ikon path {
        stroke: @white !important;
      }
    }
  }
}
