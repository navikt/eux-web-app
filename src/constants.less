@import './variabler';

@black: #000000;
@font-color: @black;
