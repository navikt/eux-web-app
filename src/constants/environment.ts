export default import.meta.env.NODE_ENV
export const IS_DEVELOPMENT = import.meta.env.NODE_ENV === 'development'
export const IS_PRODUCTION = import.meta.env.NODE_ENV === 'production'
export const IS_TEST = import.meta.env.NODE_ENV === 'test'

export const IS_Q = window.location.hostname.endsWith('preprod.local')
export const IS_P = window.location.hostname.endsWith('adeo.no')

export const APP_VERSION = import.meta.env.VITE_APP_VERSION
export const APP_BUILD_DATETIME = import.meta.env.VITE_APP_BUILD_DATETIME
export const APP_BUILD_VERSION = import.meta.env.VITE_APP_BUILD_VERSION
export const APP_BRANCH_NAME = import.meta.env.VITE_APP_BRANCH_NAME
export const APP_EESSI_KODEVERK = import.meta.env.VITE_APP_EESSI_KODEVERK
export const APP_REACT_LIB = import.meta.env.VITE_APP_REACT_LIB
