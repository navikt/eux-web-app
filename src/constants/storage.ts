export const STORAGE_SVARSED = 'svarsed'
export const STORAGE_PDU1 = 'pdu1'
