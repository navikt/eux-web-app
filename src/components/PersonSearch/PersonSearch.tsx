import Alert, { AlertStatus } from 'components/Alert/Alert'
import WaitingPanel from 'components/WaitingPanel/WaitingPanel'
import * as types from 'constants/actionTypes'
import { Person } from 'declarations/types'
import { ValidationPropType } from 'declarations/types.pt'
import _ from 'lodash'
import { Knapp } from 'nav-frontend-knapper'
import { Input } from 'nav-frontend-skjema'
import PT from 'prop-types'
import React, { useCallback, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components'
import PersonCard from '../PersonCard/PersonCard'

export interface PersonSearchProps {
  alertStatus: string | undefined;
  alertMessage: string | undefined;
  alertType: string | undefined;
  className?: string;
  initialFnr: any;
  gettingPerson: boolean;
  onAlertClose: () => void;
  onFnrChange?: () => void;
  onPersonFound? : (person: Person) => void;
  onSearchPerformed: (fnr: any) => void;
  onPersonRemoved: () => void;
  person?: Person;
  resetAllValidation: () => void;
  validation: any;
}

const AlertstripeDiv = styled.div`
  margin: 0.5rem;
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
  width: 50%;
`
const PersonSearchDiv = styled.div`
  margin-bottom: 2em;
`
const PersonSearchPanel = styled.div`
  display: flex;
  flex-direction: row;
`
const PersonSearchInput = styled(Input)`
  min-width: 24.5em;
`
const Button = styled(Knapp)`
  display: flex;
  flex: 0;
  height: 2.4em;
  align-self: flex-start;
  margin: 1.9em 0 0 1em;
 `

const PersonSearch: React.FC<PersonSearchProps> = ({
  alertStatus, alertMessage, alertType, className, initialFnr, gettingPerson,
  onAlertClose, onFnrChange, onPersonFound, onPersonRemoved, onSearchPerformed, person,
  resetAllValidation, validation
}: PersonSearchProps): JSX.Element => {
  const { t } = useTranslation()
  const [_fnr, setFnr] = useState<string | undefined>(initialFnr)
  const [_person, setPerson] = useState<Person | undefined>(undefined)
  const [localValidation, setLocalValidation] = useState<string |undefined>(undefined)

  const isPersonValid = useCallback(
    (person: Person) => (person?.fornavn?.length !== undefined && person?.fnr !== undefined)
    , []
  )

  useEffect(() => {
    if (person && !_person && isPersonValid(person)) {
      setPerson(person)
      if (_.isFunction(onPersonFound)) {
        onPersonFound(person)
      }
    }
  }, [person, _person, isPersonValid, onPersonFound])

  const sokEtterPerson = (): void => {
    const fnrPattern = /^[0-9]{11}$/
    if (_fnr && !fnrPattern.test(_fnr)) {
      setLocalValidation(t('ui:validation-invalidFnr'))
      return
    }
    setLocalValidation(undefined)
    setPerson(undefined)
    if (_.isFunction(onSearchPerformed)) {
      onSearchPerformed(_fnr)
    }
  }

  const onChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setLocalValidation(undefined)
    resetAllValidation()
    if (_.isFunction(onFnrChange)) {
      onFnrChange()
    }
    setFnr(e.target.value.trim())
  }

  const onRemovePerson = (): void => {
    setLocalValidation(undefined)
    resetAllValidation()
    setPerson(undefined)
    if (_.isFunction(onPersonRemoved)) {
      onPersonRemoved()
    }
  }

  return (
    <PersonSearchDiv>
      <PersonSearchPanel>
        <PersonSearchInput
          data-testid='personsok__input-id'
          label={t('ui:form-searchUser')}
          value={_fnr || ''}
          onChange={onChange}
          feil={validation.fnr || localValidation}
        />
        <Button onClick={sokEtterPerson} disabled={gettingPerson}>
          {gettingPerson ? <WaitingPanel size='S' message={t('ui:form-searching')} oneLine /> : t('ui:form-search')}
        </Button>
      </PersonSearchPanel>
      {alertMessage && alertType === types.SAK_PERSON_GET_FAILURE && (
        <AlertstripeDiv>
          <Alert
            type='client'
            fixed={false}
            message={t(alertMessage)}
            status={alertStatus as AlertStatus}
            onClose={onAlertClose}
          />
        </AlertstripeDiv>
      )}
      {person && isPersonValid(person) && <PersonCard className='neutral' person={person} onRemoveClick={onRemovePerson} />}
    </PersonSearchDiv>
  )
}

PersonSearch.propTypes = {
  className: PT.string,
  onFnrChange: PT.func,
  onPersonFound: PT.func,
  resetAllValidation: PT.func.isRequired,
  validation: ValidationPropType.isRequired
}

export default PersonSearch
