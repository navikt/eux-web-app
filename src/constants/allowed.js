import {IS_DEVELOPMENT} from "constants/environment.ts";

export const ALLOWED_TO_FILL_OUT = IS_DEVELOPMENT ? ["H001", "F001"] : ["H001"]
export const ALLOWED_SAKSHANDLINGER = IS_DEVELOPMENT ?
    ["H001", "X001", "X009", "F002", "F004", "F022", "F023", "F026", "F027", "F016", "R001", "R003", "R004", "Delete_Case"] :
    ["H001", "X001", "X009", "F002", "Delete_Case"]

export const HIDDEN_SAKSHANDLINGER = ["multipleParticipants", "singleParticipant"]
export const ALLOWED_SED_HANDLINGER = ["H002", "F002", "X008", "X010", "X011", "X012", ]
export const ALLOWED_SED_EDIT_AND_UPDATE = ["H001", "H002", "F002", "X008", "X010", "X011", "X012"]
