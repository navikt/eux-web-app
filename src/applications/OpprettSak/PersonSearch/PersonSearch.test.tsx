import * as types from 'constants/actionTypes'
import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import samplePerson from 'resources/tests/samplePerson'
import PersonSearch, { PersonSearchProps } from './PersonSearch'

describe('components/PersonSearch/PersonSearch', () => {
  let wrapper: ReactWrapper
  const initialMockProps: PersonSearchProps = {
    alertMessage: 'message',
    alertType: types.SAK_PERSON_GET_FAILURE,
    alertTypesWatched: [types.SAK_PERSON_GET_FAILURE],
    className: 'mock-className',
    id: 'id',
    feil: undefined,
    initialFnr: '12345678901',
    gettingPerson: false,
    onFnrChange: jest.fn(),
    onPersonFound: jest.fn(),
    onPersonRemoved: jest.fn(),
    onSearchPerformed: jest.fn(),
    person: samplePerson,
    resetAllValidation: jest.fn()
  }

  beforeEach(() => {
    wrapper = mount(<PersonSearch {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Render: match snapshot', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Render: HTML structure', () => {
    expect(wrapper.exists('.personsok')).toBeTruthy()
  })
})
