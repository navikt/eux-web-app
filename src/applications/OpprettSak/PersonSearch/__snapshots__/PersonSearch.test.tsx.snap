// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PersonSearch/PersonSearch Render: match snapshot 1`] = `
.c3 {
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
  background-color: inherit !important;
  color: !important;
  border-color: !important;
}

.c3:hover:not(:disabled) {
  background-color: !important;
  color: !important;
}

.c3:hover:not(:disabled) svg {
  color: !important;
}

.c3:hover:not(:disabled) svg path {
  stroke: !important;
}

.c3:disabled {
  color: !important;
  background-color: !important;
  border-color: !important;
}

.c3:disabled svg {
  color: !important;
}

.c3:disabled svg path {
  stroke: !important;
}

.c3:disabled .knapp__spinner {
  border: 2px solid !important;
  border-top-color: !important;
  border-right-color: !important;
}

.c3 .knapp__spinner {
  border: 2px solid !important;
  border-left-color: !important;
  border-bottom-color: !important;
}

.c3:not(:disabled) svg {
  color: !important;
}

.c3:not(:disabled) svg path {
  stroke: !important;
}

.c9 {
  display: inline-block;
  margin-left: 1rem;
}

.c11 {
  display: inline-block;
  margin-left: 0.5rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  border: 1px solid lightgray;
  background: transparent;
  margin-top: 1.5rem;
  max-width: 800px;
  min-width: 400px;
  padding: 1rem;
  border-radius: 5px;
}

.c6.personNotSelected {
  border: 3px solid red;
}

.c6.personSelected {
  border: 3px solid green;
}

.c6.neutral {
  background: white;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: flex-end;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c5 {
  margin: 0.5rem;
  min-width: 50%;
}

.c0 {
  margin-bottom: 2em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  min-width: 24.5em;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  height: 2.4em;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin: 1.9em 0 0 1em;
}

<PersonSearch
  alertMessage="message"
  alertStatus="ERROR"
  alertType="SAK/PERSON/GET/FAILURE"
  alertTypesWatched={
    Array [
      "SAK/PERSON/GET/FAILURE",
    ]
  }
  className="mock-className"
  gettingPerson={false}
  initialFnr="12345678901"
  onAlertClose={[MockFunction]}
  onFnrChange={[MockFunction]}
  onPersonFound={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "etternavn": "BLYANT",
            "fdato": "1961-02-02",
            "fnr": "02026100715",
            "fornavn": "STOR",
            "kjoenn": "M",
            "relasjoner": Array [
              Object {
                "etternavn": "BLYANT",
                "fdato": "1961-04-08",
                "fnr": "02026100714",
                "fornavn": "PEN",
                "kjoenn": "K",
                "rolle": "EKTE",
              },
              Object {
                "etternavn": "BLYANT",
                "fdato": "1980-02-02",
                "fnr": "02026100713",
                "fornavn": "LITEN",
                "kjoenn": "M",
                "rolle": "BARN",
              },
            ],
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onPersonRemoved={[MockFunction]}
  onSearchPerformed={[MockFunction]}
  person={
    Object {
      "etternavn": "BLYANT",
      "fdato": "1961-02-02",
      "fnr": "02026100715",
      "fornavn": "STOR",
      "kjoenn": "M",
      "relasjoner": Array [
        Object {
          "etternavn": "BLYANT",
          "fdato": "1961-04-08",
          "fnr": "02026100714",
          "fornavn": "PEN",
          "kjoenn": "K",
          "rolle": "EKTE",
        },
        Object {
          "etternavn": "BLYANT",
          "fdato": "1980-02-02",
          "fnr": "02026100713",
          "fornavn": "LITEN",
          "kjoenn": "M",
          "rolle": "BARN",
        },
      ],
    }
  }
  resetAllValidation={[MockFunction]}
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <Styled(Input)
            label="label:søker"
            onChange={[Function]}
            value="12345678901"
          >
            <Input
              bredde="fullbredde"
              className="c2"
              label="label:søker"
              onChange={[Function]}
              value="12345678901"
            >
              <div
                className="skjemaelement c2"
              >
                <Label
                  htmlFor="520000248-02443-4985-4868-9077671400106"
                >
                  <label
                    className="skjemaelement__label"
                    htmlFor="520000248-02443-4985-4868-9077671400106"
                  >
                    label:søker
                  </label>
                </Label>
                <input
                  aria-invalid={false}
                  className="skjemaelement__input input--fullbredde"
                  id="520000248-02443-4985-4868-9077671400106"
                  onChange={[Function]}
                  type="text"
                  value="12345678901"
                />
              </div>
            </Input>
          </Styled(Input)>
          <Styled(Styled(u))
            disabled={false}
            onClick={[Function]}
          >
            <u
              className="c3 c4"
              disabled={false}
              onClick={[Function]}
            >
              <p
                autoDisableVedSpinner={false}
                className="c3 c4"
                disabled={false}
                htmlType="submit"
                inaktivKlasseVedDisabled={false}
                kompakt={false}
                mini={false}
                onClick={[Function]}
                spinner={false}
                type="standard"
              >
                <button
                  className="knapp c3 c4 knapp--standard"
                  disabled={false}
                  onClick={[Function]}
                  type="submit"
                >
                  el:button-search
                </button>
              </p>
            </u>
          </Styled(Styled(u))>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="c5"
        >
          <AlertStripe
            form="default"
            size="1.5em"
            type="advarsel"
          >
            <div
              className="alertstripe alertstripe--advarsel"
            >
              <span
                className="alertstripe__ikon"
              >
                <span
                  className="sr-only"
                >
                  advarsel
                </span>
                <Icon
                  aria-label="advarsel-ikon"
                  kind="advarsel-sirkel-fyll"
                  role="img"
                  size="1.5em"
                >
                  <svg
                    aria-label="advarsel-ikon"
                    focusable="false"
                    height="1.5em"
                    kind="advarsel-sirkel-fyll"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1.5em"
                  >
                    <g
                      fill="none"
                      fillRule="evenodd"
                    >
                      <path
                        d="M12.205-.004l-.214.002a12.225 12.225 0 0 0-8.517 3.659C1.179 5.977-.053 9.013.002 12.208c.115 6.613 5.296 11.793 11.795 11.793l.212-.002c6.726-.116 12.105-5.595 11.99-12.21C23.883 5.178 18.702-.003 12.204-.003z"
                        fill="#FFA733"
                        fillRule="nonzero"
                      />
                      <path
                        d="M12.027 19H12A1.499 1.499 0 0 1 11.973 16L12 16a1.501 1.501 0 0 1 .027 3z"
                        fill="#262626"
                      />
                      <path
                        d="M12 5a1 1 0 0 1 1 1v7a1 1 0 0 1-2 0V6a1 1 0 0 1 1-1z"
                        fill="#262626"
                        fillRule="nonzero"
                      />
                    </g>
                  </svg>
                </Icon>
              </span>
              <Normaltekst
                className="alertstripe__tekst"
                tag="div"
              >
                <TypografiBase
                  className="alertstripe__tekst"
                  tag="div"
                  type="normaltekst"
                >
                  <div
                    className="typo-normal alertstripe__tekst"
                  >
                    message
                  </div>
                </TypografiBase>
              </Normaltekst>
            </div>
          </AlertStripe>
        </div>
      </styled.div>
      <PersonCard
        className="neutral"
        onRemoveClick={[Function]}
        person={
          Object {
            "etternavn": "BLYANT",
            "fdato": "1961-02-02",
            "fnr": "02026100715",
            "fornavn": "STOR",
            "kjoenn": "M",
            "relasjoner": Array [
              Object {
                "etternavn": "BLYANT",
                "fdato": "1961-04-08",
                "fnr": "02026100714",
                "fornavn": "PEN",
                "kjoenn": "K",
                "rolle": "EKTE",
              },
              Object {
                "etternavn": "BLYANT",
                "fdato": "1980-02-02",
                "fnr": "02026100713",
                "fornavn": "LITEN",
                "kjoenn": "M",
                "rolle": "BARN",
              },
            ],
          }
        }
      >
        <Styled(Panel)
          className="neutral"
        >
          <Panel
            border={false}
            className="c6 neutral"
          >
            <div
              className="panel c6 neutral"
            >
              <styled.div>
                <div
                  className="c7"
                >
                  <styled.div>
                    <div
                      className="c8"
                    >
                      <img
                        alt="nav-man-icon"
                        height={50}
                        src="Man.png"
                        width={50}
                      />
                      <styled.div>
                        <div
                          className="c9"
                        />
                      </styled.div>
                      <div
                        data-test-id="panelheader__tittel"
                      >
                        <Undertittel
                          data-test-id="panelheader__tittel__hoved"
                        >
                          <TypografiBase
                            data-test-id="panelheader__tittel__hoved"
                            type="undertittel"
                          >
                            <h2
                              className="typo-undertittel"
                              data-test-id="panelheader__tittel__hoved"
                            >
                              STOR
                               
                              BLYANT
                            </h2>
                          </TypografiBase>
                        </Undertittel>
                        <styled.div>
                          <div
                            className="c10"
                          >
                            <div>
                              label:fnr : 02026100715
                            </div>
                            <div>
                              label:fødselsdato: 02.02.1961
                            </div>
                          </div>
                        </styled.div>
                      </div>
                    </div>
                  </styled.div>
                  <styled.div
                    size="0.5"
                  >
                    <div
                      className="c11"
                      size="0.5"
                    />
                  </styled.div>
                  <Styled(Styled(u))
                    kompakt={true}
                    onClick={[Function]}
                  >
                    <u
                      className="c3 c12"
                      kompakt={true}
                      onClick={[Function]}
                    >
                      <p
                        autoDisableVedSpinner={false}
                        className="c3 c12"
                        disabled={false}
                        htmlType="submit"
                        inaktivKlasseVedDisabled={false}
                        kompakt={true}
                        mini={false}
                        onClick={[Function]}
                        spinner={false}
                        type="standard"
                      >
                        <button
                          className="knapp c3 c12 knapp--standard knapp--kompakt"
                          disabled={false}
                          onClick={[Function]}
                          type="submit"
                        >
                          <_default
                            color="#0067C5"
                            height="20"
                            width="20"
                          >
                            <svg
                              color="#0067C5"
                              focusable="false"
                              height="20"
                              viewBox="0 0 24 24"
                              width="20"
                            >
                              <path
                                d="M3.516 3.5h16v20h-16zm4-3h8v3h-8zm-6.5 3h22M7.516 7v12m4-12v12m4-12v12"
                                fill="none"
                                stroke="#0067C5"
                                strokeLinecap="round"
                                strokeLinejoin="round"
                                strokeMiterlimit="10"
                              />
                            </svg>
                          </_default>
                          <styled.div>
                            <div
                              className="c9"
                            />
                          </styled.div>
                          <styled.div>
                            <div
                              className="c13"
                            >
                              el:button-remove
                            </div>
                          </styled.div>
                        </button>
                      </p>
                    </u>
                  </Styled(Styled(u))>
                </div>
              </styled.div>
            </div>
          </Panel>
        </Styled(Panel)>
      </PersonCard>
    </div>
  </styled.div>
</PersonSearch>
`;
